src/inject/dynamic-theme/index.ts

```diff
@@ -459,6 +464,8 @@ export function removeDynamicTheme() {
     });
     shadowRootsWithOverrides.clear();
     forEach(styleManagers.keys(), (el) => removeManager(el));
+    loadingStyles.clear();
+    storedRejectorsForLoadingLinks.clear();
```

Comment:
I feel like a bit of Encapsulation (OOP) should be here (even though it's one line): This file should know as few details as possible, so you should import some method for cleaning up something.